katcp/server.py

```diff
@@ -1079,8 +1085,14 @@ def async_reply(f):
                     # async futures is turning out to be a pain in the ass ;)
                     self.ioloop.add_callback(reply.add_done_callback, async_reply)
                     # reply.add_done_callback(async_reply)
-                    self._logger.debug("%s FUTURE OK" % (msg.name,))
-                    return done_future
+
+                    if concurrent:
+                        # Return immediately if this is a concurrent handler
+                        self._logger.debug("%s FUTURE CONCURRENT OK", msg.name)
+                        return
+                    else:
+                        self._logger.debug("%s FUTURE OK" % (msg.name,))
```

Comment:
Since your have changed the other debug this one would also be nice as.
`self._logger.debug("%s FUTURE OK", msg.name)`
And thanks for using the args on debug logs.